packages/docusaurus-theme-classic/src/theme/DocSidebar/index.js

```diff
@@ -196,12 +186,6 @@ function DocSidebar(props) {
     );
   }
 
-  if (sidebarCollapsible) {
-    sidebarData.forEach((sidebarItem) =>
-      mutateSidebarCollapsingState(sidebarItem, path),
-    );
-  }
```

Comment:
very dangerous, particularly on the server, as mutating any object of sidebarData would affect other pages